content/en/docs/components/trainer/operator-guides/extension-framework.md

```diff
@@ -0,0 +1,84 @@
++++
+title = "Kubeflow Trainer Extension Framework"
+description = "Core concepts and architecture of the Kubeflow Trainer Extension Framework"
+weight = 70
++++
+
+This guide describes the Kubeflow Trainer Extension Framework's core concepts and its four execution phases: startup, pre-execution, build, and post-execution.
+
+## Overview
+
+The Kubeflow Trainer Extension Framework is an internal mechanism that enables flexible expansion of Runtime and TrainJob combinations in the Kubeflow ecosystem. The framework provides extension points and manages component lifecycle through four phases.
+
+<img src="/docs/components/trainer/operator-guides/images/KubeflowTrainerExtensionFrameworkOverview.drawio.svg"
+  alt="Kubeflow Trainer Extension Framework Architecture"
+  class="mt-3 mb-3">
+
+### Purpose and Intended Users
+
+The primary purpose of the Kubeflow Trainer Extension Framework is to provide a flexible and extensible mechanism for managing and executing machine learning training jobs within the Kubeflow Trainer. It is designed for platform administrators who need to extend Kubeflow Trainer with their custom plugins to fit their specific requirements.
+
+### Examples
+
+To illustrate how the framework can be used, consider the following scenarios:
+1. **Custom Validation**: A user wants to add specific validation logic to ensure that certain fields in the TrainJob are correctly configured before execution.
+2. **Dynamic Resource Deployment**: A user needs to deploy Kubernetes resources dynamically based on ML framework specific requirements.
+
+## Core Concepts
+
+### Phases
+
+The Kubeflow Trainer Extension Framework follows a structured, step-by-step execution flow. Each phase represents a logical part of the workflow:
+
+- **Startup Phase**: Executes once during the initialization of the `kubeflow-trainer-controller-manager`. This phase sets up necessary internal components.
+- **PreExecution Phase**: Triggered when a TrainJob is created or updated. This phase validates and prepares the job for execution.
+- **Build Phase**: Builds and deploys the required Kubernetes resources for training jobs.
+- **PostExecution Phase**: Runs after the job has been built and executed, checking the status and applying relevant conditions to the job.
+
+### APIs and Extension Points
+
+In each phase, there are two types of components:
+1. **Internal APIs**: These APIs are used internally by the framework and cannot be extended or modified by the user.
+2. **Extension Points**: These points are exposed to the user and allow for customization through plugins that can be added to the framework.
+
+<img src="/docs/components/trainer/operator-guides/images/KubeflowTrainerExtensionFramework.drawio.svg"
```

Comment:
@IRONICBo could you update the diagram by removing the ComponentDeployer and CreatedCondition references?

Reply:
Sure, I have remved `ComponentDeployer` and `CreatedCondition` here.